### PR TITLE
[AOTI] CPU AOTI should error out if loaded on GPU

### PR DESCRIPTION
**Pull request by malfet**
And vice versa

### CODE REVIEW

File: torch/csrc/inductor/aoti_runtime/model.h [hunk 1]
```diff
@@ -105,9 +105,16 @@ class AOTInductorModelBase {
...
+    AOTI_RUNTIME_CHECK(
+        device_type_ == aoti_torch_device_type_cuda(),
+        "Can't instantiate CUDA model for " + device_str);
     if (device_idx_ == -1) {
       AOTI_RUNTIME_DEVICE_CHECK(cudaGetDevice(&device_idx_));
     }
+#else
+    AOTI_RUNTIME_CHECK(
+        device_type_ == aoti_torch_device_type_cpu(),
+        "Can't instantiate CPU model for " + device_str);
```
Comment (mikekgfb) [inline torch/csrc/inductor/aoti_runtime/model.h:117]: why can I not instantiate a CPU model on a CUDA build?  Then how do I build an aoti runner for cpu on a host that has cuda installed?  

I do presume all the operators exist?  And the fact that we prepare tensors on cpu and then ship to cuda mean we can instantuiate cpu tensors..?

**Review by mikekgfb — CHANGES_REQUESTED** [review]
Should an AOTI runner built for cuda be able to execute both CPU and CUDA DSO models?

File: torch/csrc/inductor/aoti_runtime/model.h (hunk 1 above)
Comment (mikekgfb) [reply torch/csrc/inductor/aoti_runtime/model.h:117]: I expected that for cpu dso loaded on a rubber for AOTI, all we’d have to do is change the kDevice that is used to move input values to the proper device for processing. If it’s a cpu model we can just set that to cpu and make a null move?

Comment (desertfire) [reply torch/csrc/inductor/aoti_runtime/model.h:117]: I have been chatting to @malfet offline on this. Here, we are relying on the `USE_CUDA` macro for checking, and this macro is specified when building model.so, basically an indication of whether this model.so is for CPU or GPU. In the ideal case, we should be able to run a CPU model.so with AOTIModelContainerRunnerCuda, but not the other way around. Initially I was ok with having another follow-up PR to support such a more general case, but looks like this PR already hit that in the existing unit test, `test_pytree_inputs`, although it's probably not intentional (likely that test forgot to specify tensor device).

**Pull request closed without merging**